### ISSUE TITLE
LAPACK error invalid argument #4 to LAPACK call

### ISSUE THREAD

**Issue by rieser** [JuliaLang/julia#14065]
I get this error from the REPL. Many thanks in advance!

```
julia> A = [NaN NaN ; NaN NaN]
2x2 Array{Float64,2}:
 NaN  NaN
 NaN  NaN

julia> eigvecs(A)
 ** On entry to DGEBAL parameter number  3 had an illegal value
 ** On entry to DGEHRD parameter number  2 had an illegal value
 ** On entry to DORGHRRightDORGL2 parameter number  2 had an illegal value
 ** On entry to DHSEQR parameter number  4 had an illegal value
 ** On entry to DGEBAKNPSBDGEBAL parameter number  4 had an illegal value
ERROR: ArgumentError: invalid argument #4 to LAPACK call
 [inlined code] from strings/io.jl:43
 in geevx! at linalg/lapack.jl:1742
 in eigfact! at linalg/eigen.jl:32
 in eigfact at linalg/eigen.jl:57
 in eigvecs at linalg/eigen.jl:70
 in eval at ./boot.jl:263
```

**Comment by rieser** [comment]
Note, however, that

julia> A = [NaN NaN ; 0 0]
2x2 Array{Float64,2}:
 NaN    NaN  
   0.0    0.0

julia> eigvecs(A)
2x2 Array{Float64,2}:
 1.0  NaN
 0.0  NaN

**Comment by andreasnoack** [comment]
The underlying issue is the same as #13976.

We can an should give a better error message, but why are you doing linear algebra on all these `NaN` matrices?

**Comment by rieser** [comment]
:) A similar oversight. Forget to check for a zero somewhere, divide, do
one or two other things, and you get NaN in the matrix.

**Comment by andreasnoack** [comment]
Fixed by #14224
